drv/ArrayFIFOQueue.drv

```diff
@@ -192,13 +221,55 @@ public class ARRAY_FIFO_QUEUE KEY_GENERIC implements PRIORITY_QUEUE KEY_GENERIC,
 		length = (end = size) + 1;
 		array = newArray;
 	}
-
+	
 	@Override
 	public int size() {
 		final int apparentLength = end - start;
 		return apparentLength >= 0 ? apparentLength : length + apparentLength;
 	}
+	
+	@Override
+	public KEY_GENERIC_TYPE[] TO_KEY_ARRAY()
+	{
+		final int length = size();
+#if KEYS_PRIMITIVE
+		final KEY_GENERIC_TYPE[] newArray = new KEY_GENERIC_TYPE[length];
+#else
+		final KEY_GENERIC_TYPE[] newArray = (KEY_GENERIC_TYPE[])new Object[length];
+#endif
 
+		if (start <= end) {
+			System.arraycopy(array, start, newArray, 0, end - start);
+		}
+		else {
+			System.arraycopy(array, start, newArray, 0, length - start);
+			System.arraycopy(array, 0, newArray, length - start, end);
+		}
+		return newArray;
+	}
+	
+	@Override
+	public KEY_GENERIC_TYPE[] TO_KEY_ARRAY(KEY_GENERIC_TYPE[] input)
+	{
+		final int length = size();
+		if(input == null || input.length < length)
+		{
+#if KEYS_PRIMITIVE
+			input = new KEY_GENERIC_TYPE[length];
+#else
+			input = (KEY_GENERIC_TYPE[])new Object[length];
+#endif
+		}
+		if (start <= end) {
+			System.arraycopy(array, start, input, 0, end - start);
+		}
+		else {
+			System.arraycopy(array, start, input, 0, length - start);
+			System.arraycopy(array, 0, input, length - start, end);
+		}
```

Comment:
Inconsistent brace style example (I think fastutil follows the usual approach of keeping opening brace on the same line)

Reply:
Ok, will be adjusted.